### PR TITLE
Assignament 4 - Alba Lozano

### PR DESCRIPTION
**Pull request by ALBALOZANOG**
Alba Lozano

### CODE REVIEW

**Comment by SerPablo** [comment]
Please, review exercises 7.1 7.2 and 7.3 including rdflib

File: Assignment4/AlbaLozano_22C025/task06.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+ns3 = Namespace("http://www.w3.org/2001/vcard-rdf/3.0#")
+email_address = Literal("janesmith@gmail.com")
+g.add((JaneSmith, ns3.EMAIL, email_address))
+
+full_name = Literal("Jane Smith")
+g.add((JaneSmith, ns3.FN, full_name))
+
+given_name = Literal("Jane")
+g.add((JaneSmith, ns3.gIVEN, given_name))
```
Comment (dgarijo) [inline Assignment4/AlbaLozano_22C025/task06.py:71]: it should be GIVEN. Capitalization is important

File: Assignment4/AlbaLozano_22C025/albalozano_22c025.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+"""**Task 7.5: List the entities who know at least two other entities in the graph**"""
+
+# TO DO
+# Visualize the results
+
+
+VCARD = Namespace("http://www.w3.org/2001/vcard-rdf/3.0")
+from rdflib.plugins.sparql import prepareQuery
+from rdflib import FOAF
+g.add((ns.A, FOAF.knows, ns.B))
```
Comment (dgarijo) [inline Assignment4/AlbaLozano_22C025/albalozano_22c025.py:108]: I don't understand what you are trying to do here

**Review by dgarijo — CHANGES_REQUESTED** [review]
Additional comments to the ones raised already

File: Assignment4/AlbaLozano_22C025/task07.py [hunk 3]
```diff
@@ -23,35 +23,33 @@
...
+q1=prepareQuery('''
+    SELECT distinct ?subject
+    where{
+        ?subject rdfs:subClassOf ns:LivingThing.}''',
+    initNs={"rdfs":RDFS, "ns":ns}
+)
 
 for r in g.query(q1):
-    print(r)
+  print(r)
```
Comment (dgarijo) [inline Assignment4/AlbaLozano_22C025/task07.py:37]: You are not retrieving the full subclass hierarchy of LivingThing

File: Assignment4/AlbaLozano_22C025/task07.py [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+    WHERE {
+        ?entity ns:knows ?entity1.
+        ?entity ns:knows ?entity2.
+        FILTER (?entity1 != ?entity2)
+    }
+  ''',
+  initNs = { "vcard": VCARD, "ns": ns, "RDFS": RDFS}
+)
+for r in g.query(q5):
+    print(r.entity)
```
Comment (dgarijo) [inline Assignment4/AlbaLozano_22C025/task07.py:116]: Missing all exercises in rdflib

**Review by dgarijo — CHANGES_REQUESTED** [review]
Missing exercises.

**Comment by anaigmo** [comment]
Entregado, falta algún ejercicio pero mayormente corregido, no se admiten más PR

**Pull request merged**